dfastmi/io/RiversObject.py

```diff
@@ -188,36 +169,38 @@ def _read_rivers(self):
             The name of the river configuration file (default "rivers.ini").    
         """
         self.version = "2.0"
-        self._initialize()        
-        self._initialize_advanced()
+        for branch in self.branches:
+            for reach in branch.reaches:
+                self._initialize(river_data, branch, reach)
+                self._initialize_advanced(river_data, branch, reach)       
         
+        self._verify_reaches()
         
-        for ib in range(len(self.branches)):
-            reaches = self.allreaches[ib]
-            branch = self.branches[ib]                
-            for i in range(len(reaches)):
-                reach = reaches[i]
-                
-                hydro_q = self.hydro_q[ib][i]
-                hydro_t = self.hydro_t[ib][i]
-                auto_time = self.autotime[ib][i]
-                qfit = self.qfit[ib][i]
-                self._verify_consistency_HydroQ_and_HydroT(hydro_q, hydro_t, auto_time, qfit, branch, reach)
-
-                use_tide = self.tide[ib][i]
-                tide_boundary_condition = self.tide_bc[ib][i]
-                self._verify_consistency_Hydro_and_TideBC(use_tide, hydro_q, tide_boundary_condition, branch, reach)
+    def _verify_reaches(self):
```

Comment:
feels like a responsibility of reach

Reply:
agreed, lets pick up when we refactor Reach

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112